internal/rbd/rbd_util.go

```diff
@@ -200,6 +200,30 @@ func createImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 	return nil
 }
 
+// Open the rbdVolume after it has been connected.
+func (rv *rbdVolume) open() (*librbd.Image, error) {
+	if rv.RbdImageName == "" {
+		var vi util.CSIIdentifier
+		err := vi.DecomposeCSIID(rv.VolID)
+		if err != nil {
+			err = fmt.Errorf("error decoding volume ID (%s) (%s)", rv.VolID, err)
+			return nil, ErrInvalidVolID{err}
+		}
+		rv.RbdImageName = volJournal.GetNameForUUID(rv.NamePrefix, vi.ObjectUUID, false)
+	}
+
+	ioctx, err := rv.conn.GetIoctx(rv.Pool)
+	if err != nil {
+		return nil, err
+	}
+
+	image, err := librbd.OpenImage(ioctx, rv.RbdImageName, librbd.NoSnapshot)
```

Comment:
if the image is not found on the backend we need to return `ErrImageNotFound`

Reply:
ah, right. I'll do that in `getImageInfo()`.